### DISCUSSION TITLE
[Q&A] 请问下大佬哪里有函数调用的demo或者例子

### DISCUSSION THREAD

**Discussion by scortLi** [lobehub/lobehub#1063] [Development Question] [answered]
### 🥰 需求描述

需要自定义函数调用

### 🧐 解决方案

有没有自定义函数function call调用的demo呢，谢谢

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
具体说一下？没明白你的诉求

[a comment by an automated account is not shown]

**Comment by scortLi** [comment]
@arvinxx 就是类似于openai的函数调用呢

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
openai 的函数调用你直接看官方文档不就好了么？还是说你希望了解我们在 LobeChat 里是怎么做的？

[a comment by an automated account is not shown]

**Comment by scortLi** [comment]
@arvinxx 我想了解下你们在LobeChat里面是如何实现函数调用的，有说明吗

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
目前没文档，看下 core 实现吧：

https://github.com/lobehub/lobe-chat/blob/main/src/store/chat/slices/message/action.ts#L255

我自己感觉应该写的挺清晰的。

然后整个 function call 的交互流程，可以看看这篇： https://chat-plugin-sdk.lobehub.com/guides/plugin-invoke

**Comment by scortLi** [comment]
@arvinxx 谢谢大佬，刚刚找到了

[a comment by an automated account is not shown]
